gemini/use-cases/retrieval-augmented-generation/qna_using_query_routing/utils/intent_routing.py

```diff
@@ -0,0 +1,413 @@
+
+            What is the intent of the below message?
+            MESSAGE:{text}
+            INTENT:""",  # pylint: disable=C0301:line-too-long
+            generation_config=self.genai_qna_parameters,
+        )
+
+        if response.to_dict()["candidates"][0]["finish_reason"] != 1:
+            self.logger.info(
+                "classify_intent: No response from QnA due to LLM safety checks."
+            )
+            self.logger.info("LLM error code: %s\n", response.raw_prediction_response)
+
+        intent = response.text
+        return (
+            str(intent).replace('"', "").replace("'", "").replace("INTENT:", "").strip()
+        )
+
+    def ask_codey(self, text: str) -> str:
+        """
+        Generates code in response to code-related questions ('WRITE_CODE' intent).
+
+        Provides instructions to the chat model, handles potential errors (e.g., unclear questions, unsupported programming languages), and formats the generated code.
+
+        Args:
+            text (str): The user's code-related query.
+
+        Returns:
+            str: The generated code (or an error message if applicable).
+        """
+
+        unable_to_understand_question = self.config["error_msg"][
+            "unable_to_understand_question"
+        ]
+        non_programming_question_error_msg = self.config["error_msg"][
+            "non_programming_question_error_msg"
+        ]
+        enabled_programming_language = self.config["default"][
+            "enabled_programming_language"
+        ]
+        default_language = self.config["default"]["default_language"]
+        chat = self.model.start_chat()
+        response = chat.send_message(
+            f"""
+        You are genai Programming Language Learning Assistant.
+        Your task is to undersand the question and write a code for same.
+
+        Instructions:
+        1. If programming language is not mentioned, then use {default_language} as default programming language to write a code.
+        2. Strictly follow the instructions mentioned in the question.
+        3. If the question is not clear then you can answer "{unable_to_understand_question}"
+        4. Strictly answer the question if only {enabled_programming_language} is mentioned in question.
+
+        If the question is about other programming language then DO NOT provide any answer, just say "{non_programming_question_error_msg}"
+
+        """
+        )
+
+        response = chat.send_message(
+            f"""{text}""", generation_config=self.genai_chat_parameters
+        )
+        if response.to_dict()["candidates"][0]["finish_reason"] != 1:
+            self.logger.info(
+                "ask_codey: No response from QnA due to LLM safety checks."
+            )
+            self.logger.info("LLM error code: %s\n", response.raw_prediction_response)
+        response = response.text
+        response = response.replace("```", "\n\n```")
+        response = response.replace("```java", "``` java")
```

Comment:
The default language is Python, why is this replacement needed for Java? Also, why is it needed in general, ```language is the correct markdown format.

Reply:
Previously used LLM model response was handled using this code lines. Removed the highlighted code lines as not required when using Gemini model.